pkg/crypto/certloader/server.go

```diff
@@ -134,3 +137,15 @@ func (c *WatchedServerConfig) ServerConfig(base *tls.Config) *tls.Config {
 		MinVersion: tls.VersionTLS13,
 	}
 }
+
+// appendH2Proto appends h2 to the list of protocols if it's not already there.
+func appendH2Proto(existingProtocols []string) []string {
+	for _, p := range existingProtocols {
+		if p == alpnProtocolH2 {
+			return existingProtocols
+		}
+	}
+	ret := make([]string, 0, len(existingProtocols)+1)
+	ret = append(ret, existingProtocols...)
+	return append(ret, alpnProtocolH2)
+}
```

Comment:
nitpicking: reading only the docstring and func name I would expect `appendH2Proto` to append to the given `existingProtocols` while the implementation return a copy. I don't mind either way, but consistency between the documentation and implementation would be nice.

Reply:
Sounds good, I think it is cleaner indeed to avoid mutating the provided base `tls.Config` :+1:

Reply:
Done. I have updated the method name to `constructWithH2ProtoIfNeed`.

Reply:
Since this comment has been fixed, I have close this conversation :)